portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +36,57 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Chris!</h1>");
+    // Prepare query.	 
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    int maxComments = Integer.parseInt(request.getParameter("maxComments"));		
+    
+    // Get desired number of comments.		
+    List<String> comments = new ArrayList<>();
+    int currComments = 0;
+    for (Entity entity : results.asIterable()) {
+      if (currComments < maxComments) {
+        String words = (String) entity.getProperty("words");
+        comments.add(words);
+        currComments++;
+      }
+      else {
+        break;
+      }  				
+    }
+
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
+  }
+
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String comment = request.getParameter("text-input");
+    long timestamp = System.currentTimeMillis();
+    
+    // Create new Entity
+    Entity taskEntity = new Entity("Comment");
+    taskEntity.setProperty("words", comment);
+    taskEntity.setProperty("timestamp", timestamp);
+
+    // Add it to Datastore 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+
+    response.sendRedirect("/index.html");
+  }
+
+   /**
+   * Converts an ArrayList into a JSON string using the Gson library. 
+   */
+  private String convertToJsonUsingGson(ArrayList list) {
+    Gson gson = new Gson();
+    String json = gson.toJson(list);
+    return json;
```

Comment:
Are we still using this method? Let's delete it if not.

Reply:
No, deleted